#Task 1 – For Loop in Python

1. **Print numbers 1 to 10**
   Write a `for` loop to print all numbers from **1 to 10**.

In [ ]:
for a in range(1,11):
  print(a,end=' ')

1 2 3 4 5 6 7 8 9 10 

2. **Print even numbers**
   Use a `for` loop to print only **even numbers between 1 and 20**.

In [ ]:
a=['enter the no\t']
for b in range(2,20,2):
 print(b)

2
4
6
8
10
12
14
16
18


3. **Sum of first N natural numbers**
   Write a program that asks the user for a number `N` and prints the **sum of numbers from 1 to N** using a `for` loop.

In [ ]:
s=1
N=int(input('enter an no\t'))
for a in range(1,N+1):
 s= a+N
print(s)

enter an no	3
6


4. **Squares of numbers**
   Use a `for` loop to print the **square of numbers from 1 to 10**.


In [ ]:

for b in range(1,11):
  print(f'squar of{b}={b**2}')

squar of1=1
squar of2=4
squar of3=9
squar of4=16
squar of5=25
squar of6=36
squar of7=49
squar of8=64
squar of9=81
squar of10=100


5. **Characters in a string**
   Write a program to print each character of the string `"PYTHON"` using a `for` loop.


In [ ]:
y='python'
for word in y:
  print(word,end=",")

p,y,t,h,o,n,

6. **List elements**
   Given a list:

```python
marks = [45, 67, 89, 32, 76]
```

Write a program to print all marks greater than **50** using a `for` loop.

In [ ]:
A=[45,67,89,32,76]
for a in A:
 if a>50:
  print(a,end=' ')


67 89 76 

7. **Reverse order printing**
   Use a `for` loop with `range()` to print numbers from **10 down to 1**.

In [ ]:
for n in range(10,0,-1):
  print(n)

10
9
8
7
6
5
4
3
2
1


8. **Multiplication table of a number**
   Ask the user for a number and print its **multiplication table up to 10** using a `for` loop.


In [ ]:
s=int(input('enter an no\t'))
for c in range(1,11):
 print(f'{s}x{c}={s*c}')

enter an no	34
34x1=34
34x2=68
34x3=102
34x4=136
34x5=170
34x6=204
34x7=238
34x8=272
34x9=306
34x10=340


9. **Factorial using for loop**
   Write a program to calculate the **factorial of a given number** using a `for` loop.


In [ ]:
q=int(input('enter your number'))
f=1
for w in range(2,q +1):
  f=w*f
print(f)

enter your number6
720


10. **Nested for loop – Pattern printing**
    Write a program to print the following pattern using nested `for` loops:

```
*
* *
* * *
* * * *
* * * * *
```


In [ ]:
for z in range(5):

  for i in range(z):
    print("*",end=" ")
  print()






* 
* * 
* * * 
* * * * 



#Task 2 – While Loop in Python

1. **Print numbers 1 to 10**
   Write a `while` loop to print all numbers from **1 to 10**.


In [ ]:
b=0
while b<=9:
  b+=1
  print(b)


1
2
3
4
5
6
7
8
9
10


2. **Countdown timer**
   Use a `while` loop to print numbers from **10 down to 1**.


In [ ]:
a=10
while a>1:
 print(a)
 a-=1

10
9
8
7
6
5
4
3
2


3. **Sum of first N natural numbers**
   Write a program that asks the user for a number `N` and calculates the **sum of numbers from 1 to N** using a `while` loop.

In [ ]:
f=int(input('enter an number\t'))
d=1
s=0
while d<=f:
  s=d+s
  d=d+1
print(s)

enter an number	4
10


4. **Print even numbers**
   Use a `while` loop to print only **even numbers between 1 and 20**.

In [ ]:
a=-2
while a<=18:
  a+=2
  print(a)

0
2
4
6
8
10
12
14
16
18
20


5. **Password check**
   Write a program that keeps asking the user to enter the correct password (`"python123"`) until they get it right.


In [ ]:
p=input('enter your password\t')
q='python123'
while q!=p:
 p=input('enter your correct password\t')
print('correct password')


enter your password	123
enter your correct password	233
enter your correct password	22
enter your correct password	python123
correct password


6. **Reverse a number**
   Use a `while` loop to reverse the digits of a given number.
   *(Example: Input → 1234, Output → 4321)*


In [ ]:
q=int(input('enter an number\t'))
while q!=0:
  e=q%10
  q=q//10
  print(e,end="")

enter an number	1234
4321

7. **Digits count**
   Write a program that counts how many digits are in a number using a `while` loop.


In [ ]:
w=int(input('enter an number'))
q=0
while w>0:
 w=w//10
 q=q+1
print('digit',q)

enter an number22
digit 2


8. **Factorial using while loop**
   Write a program to calculate the **factorial of a number** using a `while` loop.


In [ ]:
 #factoryal
e=int(input('enter your number\t'))
r=1
for t in range(2,e+1):
  r=t*r
  print(r)

enter your number	5
2
6
24
120


9. **Sum of digits**
   Use a `while` loop to calculate the sum of digits of a given number.
   *(Example: 123 → 1+2+3 = 6)*

In [ ]:
a=int(input("enter an number"))
s=0
w=1
while w<=a:
  s+=w
  w+= 1

print(s)






enter an number6
21


10. **Guess the number game**
    Write a program where the computer has a secret number (say 7).
    Keep asking the user to guess until they get it right, and print how many attempts it took.


In [ ]:
a=9
s=int(input('enter an number\t'))
while a!=s:
  s =int(input('wrong guess, enter again\t'))
print("correct guess")


enter an number	2
wrong guess, enter again	3
wrong guess, enter again	9
correct guess
